src/Components/SuperAdmins/Form/index.jsx

```diff
@@ -1,99 +1,183 @@
 import React, { useState, useEffect } from 'react';
+import { useParams, useHistory } from 'react-router-dom';
 import styles from './form.module.css';
+import Button from '../../Shared/Button';
+import { Input } from '../../Shared/Inputs';
+import ConfirmModal from '../../Shared/ConfirmModal';
+import ResponseModal from '../../Shared/ResponseModal';
 
-const Form = ({ addItem, updatingItem, showForm, putItem, getSuperadmins }) => {
-  const { name, lastName, email } = updatingItem;
+const SuperAdminsForm = () => {
+  const [showConfirmAdd, setShowConfirmAdd] = useState(false);
+  const [superadminToAdd, setSuperadminToAdd] = useState(null);
+  const [showConfirmEdit, setShowConfirmEdit] = useState(false);
+  const [idToEdit, setIdToEdit] = useState(null);
+  const [editedSuperadmin, setEditedSuperadmin] = useState(null);
+  const [showModal, setshowModal] = useState(false);
   const [superadmin, setSuperadmin] = useState({});
-  useEffect(() => {
+  const history = useHistory();
+  const goBackHandle = () => {
+    history.goBack();
+  };
+  const { id } = useParams();
+
+  const closeModal = () => {
+    setshowModal(false);
+  };
+
+  const confirmAdd = async (newSuperadmin) => {
+    setShowConfirmAdd(true);
+    setSuperadminToAdd(newSuperadmin);
+  };
+  const addItem = async () => {
+    try {
+      const res = await fetch(`${process.env.REACT_APP_API_URL}/api/super-admins`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(superadminToAdd)
+      });
+      setShowConfirmAdd(false);
+      goBackHandle();
+      if (res.ok) {
+        sessionStorage.setItem('state', 'success');
+        sessionStorage.setItem('resMessage', 'New superadmin created');
+      } else {
+        sessionStorage.setItem('state', 'fail');
+        sessionStorage.setItem('resMessage', 'Failed to create superadmin');
+      }
+    } catch (error) {
+      sessionStorage.setItem('state', 'fail');
+      sessionStorage.setItem('resMessage', 'Failed to create superadmin');
+    }
+  };
+
+  const confirmEdit = (id, updatedSuperadmin) => {
+    setShowConfirmEdit(true);
+    setIdToEdit(id);
+    setEditedSuperadmin(updatedSuperadmin);
+  };
+  const putItem = async () => {
+    try {
+      const res = await fetch(`${process.env.REACT_APP_API_URL}/api/super-admins/${idToEdit}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(editedSuperadmin)
+      });
+      setShowConfirmEdit(false);
+      goBackHandle();
```

Comment:
this could be fine only if the response is a success response, to maintain a consistency